build/build-release.sh

```diff
@@ -0,0 +1,233 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -o pipefail
+set -e
+set -x
+
+SKIP_GPG=${SKIP_GPG:-false}
+
+exit_with_usage() {
+  local NAME=$(basename $0)
+  cat <<EOF
+Usage: $NAME <source|binary>
+
+Top level targets are:
+  source: Create source release tarball
+  binary: Create binary release tarball
+
+All other inputs are environment variables:
+
+RELEASE_VERSION - Release version, must match pom.xml (e.g. 1.3.0-incubating)
+SKIP_GPG        - (optional) Default false
+EOF
+  exit 1
+}
+
+PROJECT_DIR="$(
+  cd "$(dirname "$0")/.."
+  pwd
+)"
+DIST_DIR="$PROJECT_DIR/dist"
+NAME="incubating-bin"
+
+MVN="$PROJECT_DIR/build/mvn"
+
+SHASUM="sha512sum"
+if [ "$(uname)" == "Darwin" ]; then
+  SHASUM="shasum -a 512"
+fi
+
+package_binary() {
```

Comment:
Emm... why not modify make dist directly?

Reply:
Make distribution accepts params to build specific Celeborn build. Release build takes no params.

Reply:
this cause duplication building script, and is hard to maintain

Reply:
The release build should use `make-distribution.sh` with specific parameters

Reply:
OK, I'll remove it now.

Reply:
I think those logic are different unless we want to build multiple profiles at one time.

Reply:
I think it can be identified by building flags